Arduino Uno R3/examples/CN0411_example/AD7124.cpp

```diff
@@ -0,0 +1,498 @@
+*    of one or more patent holders.  This license does not release you
+*    from the requirement that you obtain separate licenses from these
+*    patent holders to use this software.
+*  - Use of the software either in source or binary form, must be run
+*    on or directly connected to an Analog Devices Inc. component.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY
+* AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <Arduino.h>
+#include "AD7124.h"
+#include "Communication.h"
+
+/***************************************************************************//**
+* @brief Reads the value of the specified register without checking if the
+*        device is ready to accept user requests.
+*
+* @param device - The handler of the instance of the driver.
+* @param pReg - Pointer to the register structure holding info about the
+*               register to be read. The read value is stored inside the
+*               register structure.
+*
+* @return Returns 0 for success or negative error code.
+*******************************************************************************/
+int32_t AD7124_NoCheckReadRegister(struct ad7124_device *device,
+				   struct ad7124_st_reg* pReg)
+{
+	int32_t ret       = 0;
+	uint8_t buffer[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+	uint8_t i         = 0;
+	uint8_t check8    = 0;
+	uint8_t msgBuf[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+
+	if(!device || !pReg)
+		return INVALID_VAL;
+
+	/* Build the Command word */
+	buffer[0] = AD7124_COMM_REG_WEN | AD7124_COMM_REG_RD |
+		    AD7124_COMM_REG_RA(pReg->addr);
+
+	/* Read data from the device */
+	ret = SPI_Read(device->slave_select_id,
+		       buffer,
+		       ((device->useCRC != AD7124_DISABLE_CRC) ? pReg->size + 1
+			: pReg->size) + 1);
+	if(ret < 0)
+		return ret;
+
+	/* Check the CRC */
+	if(device->useCRC == AD7124_USE_CRC) {
+		msgBuf[0] = AD7124_COMM_REG_WEN | AD7124_COMM_REG_RD |
+			    AD7124_COMM_REG_RA(pReg->addr);
+		for(i = 1; i < pReg->size + 2; ++i) {
```

Comment:
is `pReg->size + 2` guaranteed to be no larger than 8 [i.e. length of msgBuf] ?

Reply:
In this case the pReg->size represents the size in bytes of any register in the device. The biggest register would be the data register which on some devices supported by this driver is 32-bit long, so 4 bytes. Read with CRC checking and appended status register which is one byte should total at 8 bytes maximum to my knowledge. I think 8 should be enough in this case.